kyaml/yaml/rnode.go

```diff
@@ -512,24 +526,57 @@ func (rn *RNode) SetAnnotations(m map[string]string) error {
 }
 
 // GetLabels gets the metadata labels field.
+// If specific labels are provided, only those are returned.
+// If no specific labels are provided, then all are returned.
 // If the field is missing, returns an empty map.
 // Use another method to check for missing metadata.
-func (rn *RNode) GetLabels() map[string]string {
+func (rn *RNode) GetLabels(labels ...string) map[string]string {
 	meta := rn.getMetaData()
 	if meta == nil {
 		return make(map[string]string)
 	}
-	return rn.getMapFromMeta(meta, LabelsField)
+	return rn.getMapFromMeta(meta, LabelsField, labels...)
 }
 
 // getMapFromMeta returns map, sometimes empty, from metadata.
-func (rn *RNode) getMapFromMeta(meta *RNode, fName string) map[string]string {
-	result := make(map[string]string)
-	if f := meta.Field(fName); !f.IsNilOrEmpty() {
-		_ = f.Value.VisitFields(func(node *MapNode) error {
-			result[GetValue(node.Key)] = GetValue(node.Value)
-			return nil
-		})
+func (rn *RNode) getMapFromMeta(meta *yaml.Node, fName string, fields ...string) map[string]string {
+	f := yNodeField(meta, fName)
+	if f == nil {
```

Comment:
We should use `IsNilOrEmpty` wherever possible. It checks more than just whether the RNode is nil, and is an important distinction. This applies in a few other places as well.

Reply:
f there is a *yaml.Node, so IsNilOrEmpty wasn't available.